src/edu/csus/ecs/pc2/ui/ShadowCompareSummaryPane.java

```diff
@@ -88,12 +88,14 @@ public void updateSummary(Map<String, ShadowJudgementInfo> currentJudgementMap)
 
                 String pc2Judgement = currentJudgementMap.get(submissionID).getShadowJudgementPair().getPc2Judgement();
                 String remoteJudgement = currentJudgementMap.get(submissionID).getShadowJudgementPair().getRemoteCCSJudgement();
-                if (pc2Judgement != null && remoteJudgement != null && (pc2Judgement.equalsIgnoreCase(remoteJudgement))) {
+                if (pc2Judgement != null && remoteJudgement != null && 
```

Comment:
It WAS related to the NPE.  The reason the NPE was being generated in the first place is because the Shadow was incorrectly doing the comparisons between the PC2 and Remote runs.  Specifically, the original logic made the erroneous assumption that all runs which PC2 has judged would already also have a judgement from the remote system.  This was always true in previous testing -- but with the NADC21 (53,000+ events), PC2 was able to get some runs judged before Kattis delivered a corresponding judgement.  As a result, any test for the PC2 judgement being `equalsIgnoreCase()` to the remote judgement was invalid if there _was no remote judgement yet_.  

Fixing this required a major restructuring of the logic which generated the tests for matches between runs, which adding a testing to see if the judgement(s) were still "pending".

Said another way, you have missed the fact that the change you asked about -- eliminating the `equalsIgnoreCase()` test -- is actually not eliminated at all-- it still appears two lines below.  The CHANGE is the ADDITION of tests for "pending" prior to checking if the judgement texts are equal.